lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -121,12 +112,16 @@ def validate_config_files
 
     private
 
+    def kubeclient_configs
+      @kubeclient_configs ||= @kubeconfig_files.map { |f| KubeConfig.read(f) if File.file?(f) }.compact
+    end
+
     def build_kubeclient(api_version:, context:, endpoint_path: nil)
       # Find a context defined in kube conf files that matches the input context by name
-      configs = config_files.map { |f| KubeConfig.read(f) }
-      config = configs.find { |c| c.contexts.include?(context) }
+      raise TaskConfigurationError, "No kubeconfig files found in #{@kubeconfig_files}" if kubeclient_configs.empty?
```

Comment:
This is new. As CI pointed out, without it, if you try to build a kubeclient without validating the config files, and one of them is missing, we would 💥 with an `Errno::ENOENT`. My thinking with the way I addressed this is:
- We don't need to be re-reading the config files every time we build a new client, hence the new memoization above
- If for whatever reason the code did not run the validation before building a client, and we have both a valid config containing the desired context and a missing file, we should go ahead and do what the user wanted. If we can't do what the user wanted, we should throw our own more helpful error, not `Errno::ENOENT`.